src/middlewares/authentication/base/authentication.middleware.ts

```diff
@@ -1,34 +1,14 @@
-import admin from 'firebase-admin';
 import { Request, Response, NextFunction } from 'express';
+import { verifyRole } from '../../../services/auth.service';
 
-import * as ERROR_MSG from '../../../constants/authErrorMessages';
-
-const AuthMiddleware = (permittedRoles: Array<String>) => (
+const AuthMiddleware = (permittedRoles: Array<string>) => async (
   req: Request,
   res: Response,
   next: NextFunction
-) => {
+): Promise<void> => {
   const { token } = req.body;
 
-  admin
-    .auth()
-    .verifyIdToken(token)
-    .then(claims => {
-      const claimsKeyArr = Object.keys(claims);
-
-      if (claims != null) {
-        for (let i = 0; i < claimsKeyArr.length; i += 1) {
-          if (permittedRoles.includes(claimsKeyArr[i])) {
-            return next();
-          }
-        }
-
-        return next(new Error(ERROR_MSG.NO_PERMITTED_ROLES));
-      }
-    })
-    .catch(err => {
-      return next(err);
-    });
+  await verifyRole(token, permittedRoles, next);
```

Comment:
don't pass in next

Reply:
instead you can have verifyRole return a boolean indicating success and act on that

Reply:
Oof sorry, I'll make it so that it returns true if user is authenticated and returns false if user is unauthenticated.